tests/operator/operator.go

```diff
@@ -2407,20 +2408,56 @@ spec:
 			}
 		})
 
+		// Best effort to speed up this test. The problem is that cdi changes triggers shutdown
+		// of virt-api. The Pod is then restarted and the shutdown is counted towards failures.
+		// This triggeres a backoff which is at max 5 minutes.
+		deleteApiPods := func() {
+			pods, err := virtClient.CoreV1().
+				Pods(flags.KubeVirtInstallNamespace).
+				List(context.Background(), metav1.ListOptions{
+					LabelSelector: fmt.Sprintf("%s=%s", v1.AppLabel, "virt-api"),
+				})
+			Expect(err).ToNot(HaveOccurred())
+			for _, pod := range pods.Items {
+				for _, status := range pod.Status.ContainerStatuses {
+					if status.RestartCount != 0 {
+						err := virtClient.CoreV1().Pods(flags.KubeVirtInstallNamespace).
+							Delete(context.TODO(), pod.Name, metav1.DeleteOptions{})
+						Expect(err).To(Succeed())
+						break
+					}
+				}
+			}
+		}
+
 		It("[test_id:3153][QUARANTINE] Ensure infra can handle dynamically detecting DataVolume Support", decorators.Quarantine, func() {
 			if !libstorage.HasDataVolumeCRD() {
 				Skip("Can't test DataVolume support when DataVolume CRD isn't present")
 			}
 			// This tests starting infrastructure with and without the DataVolumes feature gate
-			var foundSC bool
-			vm, foundSC = tests.NewRandomVMWithDataVolume(cd.DataVolumeImportUrlForContainerDisk(cd.ContainerDiskAlpine), testsuite.GetTestNamespace(nil))
-			if !foundSC {
+
+			sc, exists := libstorage.GetRWOFileSystemStorageClass()
+			if !exists {
 				Skip("Skip test when Filesystem storage is not present")
 			}
 
+			imageUrl := cd.DataVolumeImportUrlForContainerDisk(cd.ContainerDiskAlpine)
+			dataVolume := libdv.NewDataVolume(
+				libdv.WithRegistryURLSource(imageUrl),
+				libdv.WithPVC(libdv.PVCWithStorageClass(sc)),
+			)
+			vmi := libvmi.New(
+				libvmi.WithDataVolume("disk0", dataVolume.Name),
+				libvmi.WithGuestMemory("512M"),
+			)
+			vm := tests.NewRandomVirtualMachine(vmi, false)
+			libstorage.AddDataVolumeTemplate(vm, dataVolume)
+
 			running := false
 			vm.Spec.Running = &running
 
+			deleteApiPods()
```

Comment:
I think this hurts the test integrity a little bit.
Why can't we just wait for the restart count to go up, and then wait for the readiness probe?

Let me know what you think, I may be missing something as I have not started the live debugging session on this yet

Reply:
It might increase the time for the test to finish by 5 minutes because of the restart backoff.

Reply:
Is there backoff? I see it exits with `reason: Completed`

Reply:
Yes, in my observation there is. But I admit I think there should not be.

Reply:
Yeah so apparently it'll take backoff into account even for zero exit code.. but how do we reach 5 minutes? This test should at max trigger 2 restarts

Reply:
I am having a go at https://github.com/kubevirt/kubevirt/pull/10773
The test takes 3 minutes on my machine

Reply:
The additional backoffs are from tests executed before.

Reply:
I see.. I think we cannot proceed with this test where one of the steps is to issue a DELETE on the pods;
it takes a toll on the test's integrity. We cannot stay confident that the mechanism works as it is supposed to (dynamically react without needing interference)

Reply:
How to we change the integrity if we restart pods before we trigger the change?

Reply:
For example if by restarting the pods, combined with a slow informer startup, you "cheat" your way to succeeding the test

Reply:
Resulting in true negative

Reply:
We can add a negative assertion before we enable/disable CDI.

Reply:
Up to you - I think there is a trade off here between integrity and runtime, and I would personally give up runtime in this case.
Keep in mind that if we go with this PR we should get rid of the Eventually loops creating the VM and just wait for virt-api stabilization instead

Reply:
To me, this is not in the spirit of a functional test because it betrays internal details.  What happens when we update virt-api/virt-controller to not have to restart this test will be slower